src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -8698,221 +8708,1350 @@ private void VerifyParameter(CSharpCompilation comp, int index, string expectedM
+                // (6,35): error CS1525: Invalid expression term '['
+                //         System.Func<int, int> x = [My(nameof(parameter))] delegate { return 1; }
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "[").WithArguments("[").WithLocation(6, 35),
+                // (6,36): error CS0103: The name 'My' does not exist in the current context
+                //         System.Func<int, int> x = [My(nameof(parameter))] delegate { return 1; }
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "My").WithArguments("My").WithLocation(6, 36),
+                // (6,46): error CS0103: The name 'parameter' does not exist in the current context
+                //         System.Func<int, int> x = [My(nameof(parameter))] delegate { return 1; }
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "parameter").WithArguments("parameter").WithLocation(6, 46),
+                // (6,59): error CS1002: ; expected
+                //         System.Func<int, int> x = [My(nameof(parameter))] delegate { return 1; }
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "delegate").WithLocation(6, 59),
+                // (6,81): error CS1002: ; expected
+                //         System.Func<int, int> x = [My(nameof(parameter))] delegate { return 1; }
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(6, 81)
+                );
+        }
+
+        [Fact]
+        public void ParameterScope_InMethodAttributeNameOf_Delegate()
+        {
+            var source = @"
+[My(nameof(parameter))] delegate int MyDelegate(int parameter);
+
+public class MyAttribute : System.Attribute
+{
+    public MyAttribute(string name1) { }
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
+            comp.VerifyDiagnostics(
+                // (2,12): error CS0103: The name 'parameter' does not exist in the current context
+                // [My(nameof(parameter))] delegate int MyDelegate(int parameter);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "parameter").WithArguments("parameter").WithLocation(2, 12)
+                );
+
+            VerifyParameter(comp, 0, null);
+
+            comp = CreateCompilation(source, parseOptions: TestOptions.RegularNext);
+            comp.VerifyDiagnostics();
+
+            VerifyParameter(comp, 0, "System.Int32 MyDelegate.Invoke(System.Int32 parameter)");
+        }
+
+        [Fact]
+        public void ParameterScope_InMethodAttributeNameOf_Delegate_ConflictingName()
+        {
+            var source = @"
+[My(nameof(TParameter))] delegate int MyDelegate<TParameter>(int TParameter);
+
+public class MyAttribute : System.Attribute
+{
+    public MyAttribute(string name1) { }
+}
+";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
+            comp.VerifyDiagnostics();
+
+            VerifyTParameter(comp, 0, "MyDelegate<TParameter>");
+
+            source = @"
```

Comment:
>source = @" [](http://example.com/codeflow?start=12&length=11)

Is it necessary to change the value? I think it would be clearer to observe behavior change for the same source

Reply:
I think it is. There's some differences between VerifyParameter and VerifyTParameter, for instance around SymbolKind check and LookupSymbols expectations.

Reply:
Correction: I can share the same source with a minor tweak to VerifyParameter to configure the name of the parameter.